back/app/models/text_image.rb

```diff
@@ -14,16 +14,10 @@
 #  text_reference  :string           not null
 #
 class TextImage < ApplicationRecord
+  attribute :text_reference, :string, default: -> { SecureRandom.uuid }
```

Comment:
Make sure the `text_reference` is initialized right away, without waiting for validation.